fe/fe-core/src/main/java/com/starrocks/analysis/FunctionCallExpr.java

```diff
@@ -962,4 +964,56 @@ public boolean isSelfMonotonic() {
         }
         return false;
     }
+
+    // geo function whose name starts_with 'st_' and is not 'st_astext' generates invalid utf8 strings. so it cannot
+    // be used as a argument to a string-typed parameter of a function.
+    // 1. select reverse(st_circle(...)) unacceptable
+    // 2. select reverse(st_astext(st_circle(...)) acceptable
+    // 3. select reverse(cast(st_circle(...) as varchar)) unacceptable
+    // 4. select reverse(cast(cast(st_circle(...) as varchar) as varchar))) unacceptable
+    public static void checkGeoFunctionGeneratedInvalidUtf8(FunctionCallExpr node) throws SemanticException {
```

Comment:
FunctionCallExpr.java is treated as a defined structure in the new optimizer, no new logic code will be added, and existing code will also be deleted. The new function analyze logic code should be placed in FunctionAnalyzer

Reply:
ok